Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -476,25 +477,47 @@ RefPtr<StyleRuleBase> CSSParserImpl::consumeAtRule(CSSParserTokenRange& range, A
 // https://drafts.csswg.org/css-syntax/#consume-a-qualified-rule
 RefPtr<StyleRuleBase> CSSParserImpl::consumeQualifiedRule(CSSParserTokenRange& range, AllowedRules allowedRules)
 {
+    const auto initialRange = range;
+
     auto isNestedStyleRule = [&] {
         return isNestedContext() && allowedRules <= AllowedRules::RegularRules;
     };
 
     const CSSParserToken* preludeStart = &range.peek();
 
     // Parsing a selector (aka a component value) should stop at the first semicolon (and goes to error recovery)
-    // instead of consuming the whole list of declaration (in nested context).
+    // instead of consuming the whole list of declarations (in nested context).
     // At top level (aka non nested context), it's the normal rule list error recovery and we don't need this.
     while (!range.atEnd() && range.peek().type() != LeftBraceToken && (!isNestedStyleRule() || range.peek().type() != SemicolonToken))
         range.consumeComponentValue();
 
     if (range.atEnd())
-        return nullptr; // Parse error, EOF instead of qualified rule block
+        return { }; // Parse error, EOF instead of qualified rule block
 
     // See comment above
     if (isNestedStyleRule() && range.peek().type() == SemicolonToken) {
         range.consume();
-        return nullptr;
+        return { };
+    }
+
+    // https://github.com/w3c/csswg-drafts/issues/9336#issuecomment-1719806755
```

Comment:
Does this comment add much value?

Reply:
I think so (it's not trivial to understand why this patch otherwise)